File: 03.LongestSubstringWithoutRepeatingCharacters/main.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int length = s.length();
        int maxLength = 0;

        for (int i = 0; i < length; i++) {
            List<Character> myArr = new ArrayList<>();
            for (int j = i; j < length; j++) {
                if (myArr.contains(s.charAt(j))) {
                    break;
                } else {
                    myArr.add(s.charAt(j));
                }
            }
            if (myArr.size() > maxLength) {
                maxLength = myArr.size();
            }
        }
        return maxLength;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();

        System.out.println(solution.lengthOfLongestSubstring("abcabcbb"));
        System.out.println(solution.lengthOfLongestSubstring("bbbbb"));
        System.out.println(solution.lengthOfLongestSubstring("pwwkew"));
    }
}

File: 03.LongestSubstringWithoutRepeatingCharacters/main.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        int length = s.Length;
        int maxLength = 0;

        for (int i = 0; i < length; i++)
        {
            List<char> arr = new List<char>();
            for (int j = i; j < length; j++)
            {
                if (arr.Contains(s[j]))
                {
                    break;
                }
                else
                {
                    arr.Add(s[j]);
                }
            }

            if (arr.Count > maxLength)
            {
                maxLength = arr.Count;
            }
        }

        return maxLength;
    }
}

public class Program
{
    public static void Main()
    {
        Solution solution = new Solution();

        Console.WriteLine(solution.LengthOfLongestSubstring("abcabcbb"));
        Console.WriteLine(solution.LengthOfLongestSubstring("bbbbb"));
        Console.WriteLine(solution.LengthOfLongestSubstring("pwwkew"));
    }
}
